### PR TITLE
Runnable identifier

### PR DESCRIPTION
**Pull request by richtja**
This commit introduces a way how to explicitly specify runnable identifier through Job API or runnable recipe. This change shouldn't affect current behaviour and if the identifier is not specified it will be generated based on the `runner.identifier_format` config variable.

Reference: #5964

### CODE REVIEW

**Review by clebergnu — CHANGES_REQUESTED** [review]
Hi @richtja,

This LGTM, but I will ask you to consider:

* Adding one example of the identifier to the [recipes](https://avocado-framework.readthedocs.io/en/latest/guides/writer/chapters/recipes.html) documentation section
* Adding identifier example to `examples/nrunner/recipes/runnable` or `examples/nrunner/recipes/runnables`


One other issue that this opens is what to present on an `avocado list` command.  What I mean is that, until now, it was presenting the "only" information available, that is, the URI.  Example:

```
avocado list examples/nrunner/recipes/runnable/exec_test_sleep_3.json 
exec-test /bin/sleep
```

Now, with a change such as:

```diff
diff --git a/examples/nrunner/recipes/runnable/exec_test_sleep_3.json b/examples/nrunner/recipes/runnable/exec_test_sleep_3.json
index 23113758b..8e4284eb0 100644
--- a/examples/nrunner/recipes/runnable/exec_test_sleep_3.json
+++ b/examples/nrunner/recipes/runnable/exec_test_sleep_3.json
@@ -1 +1 @@
-{"kind": "exec-test", "uri": "/bin/sleep", "args": ["3"]}
+{"kind": "exec-test", "uri": "/bin/sleep", "args": ["3"], "identifier": "sleep-3-seconds"}
```

It may make sense to show, instead:

```
avocado list examples/nrunner/recipes/runnable/exec_test_sleep_3.json 
exec-test sleep-3-seconds
```

**Comment by richtja** [comment]
Hi @clebergnu, thanks for your review. I agree with you, and I will update the examples. Also, it reminded to me that I should update our documentation as well.

**Comment by richtja** [comment]
Hi @clebergnu, I have added the examples and update the `avocado list`. Please have a look.

File: avocado/schemas/runnable-recipe.schema.json [hunk 1]
```diff
@@ -28,6 +28,10 @@
         "config": {
             "description": "Avocado settings that should be applied to this runnable.  At least the ones declared as CONFIGURATION_USED in the runner specific for this kind should be present",
             "type": "object"
+        },
+        "identifier": {
+            "description": "ID of runnable which will be used for indetification during the runtime and in logs. If this is not specified it will be autogenerated based on `runner.identifier_format` config value",
```
Comment (clebergnu) [inline avocado/schemas/runnable-recipe.schema.json:33]: typo: s/indetification/identification/

File: examples/nrunner/recipes/runnable/identifier.json [hunk 2]
```diff
@@ -0,0 +1 @@
+{"kind": "noop", "identifier": "1-test-noop"}
```
Comment (clebergnu) [inline examples/nrunner/recipes/runnable/identifier.json:1]: I think having the example identifier here starting with "1-" makes it easy to confuse that with a [Test ID](https://avocado-framework.readthedocs.io/en/latest/guides/user/chapters/concepts.html#test-id)'s first component.

**Review by clebergnu — CHANGES_REQUESTED** [review]
Hi @richtja,

I've spotted two things for you to address.  Thanks for this version!

**Comment by richtja** [comment]
Hi @clebergnu, thanks for your review. I have addressed your comments via force-push. Please have a look.

**Comment by richtja** [comment]
Hi @clebergnu, I have just updated the `examples/jobs/custom_exec_test.py`. Please have a look.

**Review by clebergnu — APPROVED** [review]
LGTM, thanks!

**Pull request merged**